src/editors/autocompleteEditor.js

```diff
@@ -15,316 +15,466 @@ import {
 } from './../helpers/dom/element';
 import HandsontableEditor from './handsontableEditor';
 
-const AutocompleteEditor = HandsontableEditor.prototype.extend();
+const privatePool = new WeakMap();
 
 /**
  * @private
  * @editor AutocompleteEditor
  * @class AutocompleteEditor
  * @dependencies HandsontableEditor
  */
-AutocompleteEditor.prototype.init = function(...args) {
-  HandsontableEditor.prototype.init.apply(this, args);
-  this.query = null;
-  this.strippedChoices = [];
-  this.rawChoices = [];
-};
+class AutocompleteEditor extends HandsontableEditor {
+  constructor(instance) {
+    super(instance);
 
-AutocompleteEditor.prototype.getValue = function() {
-  const selectedValue = this.rawChoices.find((value) => {
-    const strippedValue = this.stripValueIfNeeded(value);
+    privatePool.set(this, {
+      skipOne: false,
+    });
+  }
 
-    return strippedValue === this.TEXTAREA.value;
-  });
+  /**
+   * Initializes an editor's intance.
+   */
+  init() {
+    super.init();
 
-  if (isDefined(selectedValue)) {
-    return selectedValue;
+    this.query = null;
```

Comment:
All properties should be defined in the constructor as the same as [here](https://github.com/handsontable/handsontable/pull/5628/files#diff-6c12546315d5a1a24cf343b03ff3ed89R155) (see comment).